src/shader/bsdf/bsdfSampling.glsl.js

```diff
@@ -426,35 +465,30 @@ export const bsdfSamplingGLSL = /* glsl */`
 		}
 
 		vec3 wi;
-		vec3 clearcoatWi;
 
 		float r = sobol( 15 );
 		if ( r <= cdf[0] ) { // diffuse
 
-			wi = diffuseDirection( wo, surf );
-			clearcoatWi = normalize( clearcoatInvBasis * normalize( normalBasis * wi ) );
+			wi = diffuseDirection( localWo, surf );
 
 		} else if ( r <= cdf[1] ) { // specular
 
-			wi = specularDirection( wo, surf );
-			clearcoatWi = normalize( clearcoatInvBasis * normalize( normalBasis * wi ) );
+			wi = specularDirection( localWo, surf );
 
 		} else if ( r <= cdf[2] ) { // transmission / refraction
 
-			wi = transmissionDirection( wo, surf );
-			clearcoatWi = normalize( clearcoatInvBasis * normalize( normalBasis * wi ) );
+			wi = transmissionDirection( localWo, surf );
 
 		} else if ( r <= cdf[3] ) { // clearcoat
 
-			clearcoatWi = clearcoatDirection( clearcoatWo, surf );
+			vec3 clearcoatWi = clearcoatDirection( clearcoatWo, surf );
 			wi = normalize( invBasis * normalize( clearcoatNormalBasis * clearcoatWi ) );
 
 		}
 
 		SampleRec result;
-		result.pdf = bsdfEval( wo, clearcoatWo, wi, clearcoatWi, surf, diffuseWeight, specularWeight, transmissionWeight, clearcoatWeight, result.specularPdf, result.color );
-		result.direction = wi;
-		result.clearcoatDirection = clearcoatWi;
+		result.direction = normalBasis * wi;
+		result.pdf = bsdfEval( wo, result.direction, surf, diffuseWeight, specularWeight, transmissionWeight, clearcoatWeight, result.specularPdf, result.color );
```

Comment:
This transformation into world space, passing into bsdfEval, and then back into local space loses a meaningful amount of  precision causing dark pixels.